### ISSUE TITLE
Could users installed the DIANN1.7.15 on linux system for DIAPASEF file analysis?

### ISSUE THREAD

**Issue by Mnago** [vdemichev/DiaNN#92]
I tried to download the source code of DIANN 1.7.15 and installed it on the Linux system. But the log shows that there was DIANN1.7.12 installed on the Linux system. Then, I tried to download the source code of DIANN1.7.14 and subsequently installed it on Linux. The same issue occurred on the source code of DIANN 1.7,14.  The detailed log is attached.

1. Did anything wrong with those two source codes?
2. Did the DIANN1.7.15 could be installed on the Linux system?
3. By the way, we want to use the windows command line to run DIANN 1.7.15 instead of GUI, did any difference in parameter setting?

Best regards,

Twinnie

![log](https://user-images.githubusercontent.com/52949420/1

**Comment by nathemie** [comment]
We downloaded the source code from https://github.com/vdemichev/DiaNN/releases/tag/1.7.15 and compiled the executable according to your instruction. However, the version we complied is still 1.7.2, unable to deal with timstof .d data.
We'd like to know if 1.7.5 version is supported on linux and how can we obtain the version running for DIAPASEF analysis for linux machines?

**Comment by vdemichev** [comment]
Hi All,
 
The source code on GitHub corresponds to the older version of DIA-NN (1.7.12), please see the readme.
I do plan to release a Linux build of 1.7.15 in the near future (will be a ready-to-use binary, no need to compile). It will not initially support access to .d folders directly on Linux, you'll need to convert to .dia on Windows first. But direct acccess to .d on Linux is planned, it just will take some time to implement this (at least months).

Best wishes,

Vadim

**Comment by vdemichev** [comment]
Most commands are the same in 1.7.15 and 1.7.12, so most scripts will work fine without any modification. You can always use the GUI to check if you are passing the right commands to the command line tool.

**Comment by dingxuan1990** [comment]
Thank you for your wonderful work and kindly reply. We are looking forward to the release of Linux build of 1.7.15 in the future.

**Comment by Mnago** [comment]
> Most commands are the same in 1.7.15 and 1.7.12, so most scripts will work fine without any modification. You can always use the GUI to check if you are passing the right commands to the command line tool.

Thank you, we have went through the workflow by command line just now.

**Comment by jkelabora** [comment]
> Hi All,
> 
> The source code on GitHub corresponds to the older version of DIA-NN (1.7.12), please see the readme.
> I do plan to release a Linux build of 1.7.15 in the near future (will be a ready-to-use binary, no need to compile). It will not initially support access to .d folders directly on Linux, you'll need to convert to .dia on Windows first. But direct acccess to .d on Linux is planned, it just will take some time to implement this (at least months).
> 
> Best wishes,
> 
> Vadim


Hi Vadim,

Firstly, congratulations on this project – we (@AndrewWebbWEHI and team) have been most impressed by it in our initial testing.

Secondly, apologies for commenting on a closed issue, this just seemed like the most relevant place to ask if you could provide an update on how direct access to diaPASEF .d on Linux is shaping up?

We are close to deciding on what tool to use for our high-throughput diaPASEF data processing and would love to make a case to use DIA-NN if the timelines can work.

Kind regards,
Julian

**Comment by vdemichev** [comment]
Hi Julian,

Me and Bruker have been looking into this, for now there some technical problems, but it's moving.
The way I like to think about it is that DIA-NN on a single Windows machine is still faster than alternative software on a Linux cluster :) But yes, there've been lots of requests to make things fully-functional on Linux, so we are working on this. 

Best,
Vadim

**Comment by vdemichev** [comment]
We do seem to have a DIA-NN version though which can process diaPASEF .d runs on Linux under Wine. Maybe this is how we will eventually support Linux, by just distributing a Wine-compatible executable.

**Comment by jkelabora** [comment]
Hi Vadim,

Thanks for your prompt reply. Yes, the speed of DIA-NN execution is a _major_ factor for us – it's remarkable, particularly in comparison to the alternatives.

A Wine-compatible executable would certainly do the trick, although for us, it would depend on the installation procedure. I probably should've mentioned this earlier, but we execute everything in Docker containers so we're looking to install and run DIA-NN in that manner. Linux-based would be ideal, but Windows-based can work as well.

Currently, `1.7.12` can be installed on a Linux container:
```
# (Other Linux setup commands excluded for clarity)
ADD https://github.com/vdemichev/DiaNN/archive/refs/tags/1.7.12.tar.gz .
RUN tar -xzf 1.7.12.tar.gz && rm 1.7.12.tar.gz
RUN cd /DiaNN-1.7.12/mstoolkit/ && make
```
And on a Windows container:
```
# (Other Windows setup commands excluded for clarity)
ADD https://github.com/vdemichev/DiaNN/releases/download/1.7.12/DIA-NN-Setup.msi C:/DIA-NN-Setup.msi
RUN C:/DIA-NN-Setup.msi /passive
```
The later releases, with diaPASEF capability:

1. Use `.exe` assets for Windows which can't be installed by script
2. Aren't available on Linux

Could the Wine-compatible executable be installed in a Docker container?
Could the `.exe` include a flag to allow scripted installs or could an `.msi` be provided?

I'm conscious of leading this issue down a different course, so should i raise separate issues for container support on each platform?

Also, i'm more than happy to help with this if you'd like to point me in the right direction so i can come with PRs rather than issues?

Best regards,
Julian

**Comment by vdemichev** [comment]
Hi Julian,

It's easy for me to build DIA-NN so that it does not use the extra libraries.
I will hopefully have an update on this or a proper Linux version soon.
Fine to continue in this issue. 

Best wishes,

Vadim

**Comment by jkelabora** [comment]
Hi Vadim – sounds great – we'll stay tuned!

**Comment by chambm** [comment]
Msconvert and Skyline run in a Docker container with Wine, with vendor reading support from Agilent, Bruker, Sciex, Shimadzu, Thermo, and Waters; so I'm sure DIA-NN would work too!

**Comment by vdemichev** [comment]
Wow, sounds almost magically good! 

We currently seem to have a Linux binary which supports .dia, .mzML and .d (diaPASEF) natively, and can also do deep learning spectra prediction as on Windows. We are testing it now. Getting support for Sciex and Thermo would be really cool, will see if we can make them work with Wine.

Vadim

**Comment by chambm** [comment]
Yes, magic is exactly what I was thinking when I finally got it to work. :) The most amazing thing is it works seamlessly with mixed-mode C++/CLI DLLs (which we use to wrap the .NET vendor DLLs to access them from the pwiz C++ core); using an embedded mono or .NET standard runtime couldn't do that.

**Comment by vdemichev** [comment]
OK, while we are testing the native Linux version, turns out the latest version of Wine (6.8; need to download it from the development branch) allows to run both DIA-NN.exe (the GUI) and diann.exe (the command line tool) just fine, and DIA-NN seems fully-functional.

**Comment by vdemichev** [comment]
Both DIA-NN.exe (the GUI) and diann.exe (the command line tool) seem to work fine under the latest development Wine (version 6.8). Including deep learning & dia-PASEF.

**Comment by chambm** [comment]
That's great! I suggest Dockerizing it then and maybe even making that part of your release process. Did you try reading a WIFF or RAW file? If those worked it would surprise me unless recent versions of Wine include .NET. I'm not surprised about diaPASEF because that's a native C library.

**Comment by vdemichev** [comment]
No, unfortunately neither .wiff nor .raw work. So these have to be converted on a Windows PC.

**Comment by chambm** [comment]
What error do you get? You might try inside the pwiz/Skyline container (or its parent wine-dotnet container) because that does that does have .NET installed (for 64-bit, which, at least back when I did it, was the tricky bit; 32-bit .NET was easy).

**Comment by vdemichev** [comment]
Actually turns out I just forgot to copy Sciex DLLs to the folder with DIA-NN. So yes, reading .wiff also works with Wine. With Thermo .raw (after installing MSFileReader) get this:
`Loading run /media/sf_Linux/Fig2 HeLa-0-5h_MHRM_R01_T0.raw
0024:err:ntdll:RtlpWaitForCriticalSection section 0000000000120100 "dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0024, blocked by 0118, retrying (60 sec)`

**Comment by chambm** [comment]
It's cool that WIFF works. Maybe it works with Wine's default Mono runtime because it doesn't have any mixed-mode DLLs.

You're using MSFileReader instead of RawFileReader? That would rely on COM rather than .NET interop. One of the only vendors in pwiz's Docker that doesn't work is one that also relies on COM (CompassXtractMS). I think RawFileReader is the same as the WIFF SDK: it doesn't have any mixed-mode DLLs. I bet if you switched to RawFileReader that would probably work fine on Mono as well.

**Comment by vdemichev** [comment]
There's a working Linux build now (for a development version of DIA-NN), if interested, please email me (my email is at the bottom of the first page here https://github.com/vdemichev/DiaNN/blob/master/DIA-NN%20GUI%20manual.pdf) and I will send you a link.

**Comment by vdemichev** [comment]
Now all solved in the official release of DIA-NN 1.8
